lib/runners/processor/hadolint.rb

```diff
@@ -0,0 +1,105 @@
+module Runners
+  class Processor::Hadolint < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::RunnerConfig.base.update_fields { |fields|
+        fields.merge!({
+                        target: enum?(string, array(string)),
+                        ignore: enum?(string, array(string)),
+                        trusted_registry: enum?(string, array(string)),
+                      })
+      }
+    end
+
+    DEFAULT_TARGET = "Dockerfile".freeze
+
+    def self.ci_config_section_name
+      "hadolint"
+    end
+
+    def analyzer_name
+      "hadolint"
+    end
+
+    def analyze(changes)
+      ensure_runner_config_schema(Schema.runner_config) do |config|
+        @config = config
+        run_analyzer
+      end
+    end
+
+    private
+
+    def config
+      @config or raise "Must be initialized!"
+    end
+
+    def analyzer_options
+      [].tap do |opts|
+        opts << "--format=json"
+        Array(config[:trusted_registry]).each do |trusted|
+          opts << "--trusted-registry=#{trusted}"
+        end
+        Array(config[:ignore]).each do |ignore|
+          opts << "--ignore=#{ignore}"
+        end
+        Array(config[:target] || DEFAULT_TARGET).each do |target|
+          opts << target
+        end
+      end
+    end
+
+    def run_analyzer
+      stdout, stderr, status = capture3 analyzer_bin, *analyzer_options
+      if status.exitstatus == 1 && stdout.strip == "Please provide a Dockerfile"
+        add_warning "No Docker file provided"
+        return Results::Success.new(guid: guid, analyzer: analyzer)
+      end
+
+      if stderr.present? && status.exitstatus == 1
+        return Results::Failure.new(guid: guid, analyzer: analyzer, message: stderr)
+      end
+
+      Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+        parse_result(stdout).each { |v| result.add_issue(v) }
+      end
+    end
+
+    # Output format:
+    #
+    #     { line: number, code: rule, message: description, column: number, file: filepath, level: level }
+    #
+    # Example:
+    #
+    #     {"line":14,"code":"DL3003","message":"Use WORKDIR to switch to a directory","column":1,"file":"images/hadolint/Dockerfile","level":"warning"}
+    #
+    # @see https://github.com/hadolint/hadolint#rules
+    # @param stdout [String]
+    def parse_result(stdout)
+      JSON.parse(stdout, symbolize_names: true).map do |file|
+        path = relative_path(file[:file])
+
+        line = file[:line]
+        id = file[:code]
+
+        Issue.new(
+          path: path,
+          location: Location.new(start_line: line),
+          id: id,
+          message: file[:message],
+          object: {
+            severity: file[:level],
+          },
+          links: link_to_wiki(id),
+        )
+      end
+    end
+
+    def link_to_wiki(id)
+      if id.match(/^DL/)
```

Comment:
[nitpick] I recommend using [`start_with?`](http://ruby-doc.org/core-2.6.5/String.html#method-i-start_with-3F) for more readability.

```suggestion
      if id.start_with? "DL"
```